src/platform/windows/misc.cpp

```diff
@@ -550,6 +556,243 @@ namespace platf {
+      lookup_string = extension;
+    }
+
+    // Override HKEY_CLASSES_ROOT and HKEY_CURRENT_USER to ensure we query the correct class info
+    if (!override_per_user_predefined_keys(token)) {
+      return converter.from_bytes(raw_cmd);
+    }
+
+    // Find the command string for the specified class
+    std::array<WCHAR, MAX_PATH> shell_command_string;
+    DWORD out_len = shell_command_string.size();
+    res = AssocQueryStringW(ASSOCF_NOTRUNCATE, ASSOCSTR_COMMAND, lookup_string.c_str(), L"open", shell_command_string.data(), &out_len);
+
+    // In some cases (UWP apps), we might not have a command for this target. If that happens,
+    // we'll have to launch via cmd.exe. This prevents proper job tracking, but that was already
+    // broken for UWP apps anyway due to how they are started by Windows. Even 'start /wait'
+    // doesn't work properly for UWP, so really no termination tracking seems to work at all.
+    //
+    // FIXME: Maybe we can improve this in the future.
+    if (res == HRESULT_FROM_WIN32(ERROR_NO_ASSOCIATION)) {
+      BOOST_LOG(warning) << "Using trampoline to handle target: "sv << raw_cmd;
+      std::wcscpy(shell_command_string.data(), L"cmd.exe /c start \"\" \"%1\" %*");
+      res = S_OK;
+    }
+
+    // Reset per-user keys back to the original value
+    override_per_user_predefined_keys(NULL);
+
+    if (res != S_OK) {
+      BOOST_LOG(warning) << "Failed to query command string for raw command: "sv << raw_cmd << " ["sv << util::hex(res).to_string_view() << ']';
+      return converter.from_bytes(raw_cmd);
+    }
+
+    // Finally, construct the real command string that will be passed into CreateProcess().
+    // We support common substitutions (%*, %1, %2, %L, %W, %V, etc), but there are other
+    // uncommon ones that are unsupported here.
+    //
+    // https://web.archive.org/web/20111002101214/http://msdn.microsoft.com/en-us/library/windows/desktop/cc144101(v=vs.85).aspx
+    std::wstring cmd_string { shell_command_string.data() };
+    size_t match_pos = 0;
+    while ((match_pos = cmd_string.find_first_of(L'%', match_pos)) != std::wstring::npos) {
+      std::wstring match_replacement;
+      auto next_char = std::tolower(match_pos + 1 < cmd_string.size() ? cmd_string.at(match_pos + 1) : 0);
+      switch (next_char) {
+        // No next character
+        case 0:
+          break;
+
+        // Escape character
+        case L'%':
+          // Skip this character and the next one
+          match_pos += 2;
+          continue;
+
+        // Argument replacements
+        case L'0':
+        case L'1':
+        case L'2':
+        case L'3':
+        case L'4':
+        case L'5':
+        case L'6':
+        case L'7':
+        case L'8':
+        case L'9': {
+          // Arguments numbers are 1-based, except for %0 which is equivalent to %1
+          int index = next_char - L'0';
+          if (next_char != L'0') {
+            index--;
+          }
```

Comment:
Is there a way to just check against any integer? It's probably rare, but there could be more than 10 args.

Reply:
`%0`-`%9` is actually exhaustive. The argument replacements are only a single `%` followed by a single character.

When I tested using a shell command of `cmd.exe /k echo %0 %1 %2 %3 %4 %5 %6 %7 %8 %9 %10` and opened the test via the Run dialog with `C:\users\camer\foo.shelltest 1 2 3 4 5 6 7 8 9 10 11 12`, the output of the resulting command prompt was: `C:\Users\camer\foo.shelltest C:\Users\camer\foo.shelltest 1 2 3 4 5 6 7 8 C:\Users\camer\foo.shelltest0`. The last replacement of `%10` was parsed by Windows as `%1` then `0` was appended to the end.